fixtures/dynamicTemplates.js

```diff
@@ -50,15 +53,21 @@ const dynamic_templates = [
       }
     }
   },
+  // Default all other strings not otherwise specified to keyword
   {
     strings: {
       match_mapping_type: 'string',
       mapping: { type: 'keyword' }
     }
+  },
+  {
+    numerics: {
+      match_mapping_type: 'long',
+      mapping: { type: 'float' }
+    }
```

Comment:
we now default to mapping any field that's not otherwise explicitly mapped to float instead of long, to avoid a situation where the first time the field and has a non-decimal value that it gets mapped to long, and then all other values for that field, even if they're decimal, will be truncated to longs in the index